src/expr/src/agg/mod.rs

```diff
@@ -25,48 +28,44 @@ mod def;
 // concrete aggregators
 mod approx_count_distinct;
 mod array_agg;
-mod count_star;
 mod general;
 mod jsonb_agg;
 mod mode;
 mod percentile_cont;
 mod percentile_disc;
 mod string_agg;
 
-// wrappers
-// XXX(wrj): should frontend plan these as operators?
-mod distinct;
-mod filter;
-mod orderby;
-mod projection;
-
 pub use self::def::*;
-use self::distinct::Distinct;
-use self::filter::*;
-use self::orderby::ProjectionOrderBy;
-use self::projection::Projection;
 
 /// An `Aggregator` supports `update` data and `output` result.
 #[async_trait::async_trait]
-pub trait Aggregator: Send + DynClone + 'static {
+pub trait Aggregator: Send + Sync + DynClone + 'static {
     fn return_type(&self) -> DataType;
 
-    /// `update_single` update the aggregator with a single row with type checked at runtime.
-    async fn update_single(&mut self, input: &DataChunk, row_id: usize) -> Result<()> {
-        self.update_multi(input, row_id, row_id + 1).await
-    }
+    /// Update the aggregator with multiple rows.
+    ///
+    /// This is used in streaming aggregation.
+    async fn update(&mut self, input: &StreamChunk) -> Result<()>;
+
+    /// Update the aggregator with a range of rows.
+    ///
+    /// This is used in batch aggregation.
+    async fn update_range(&mut self, input: &StreamChunk, range: Range<usize>) -> Result<()>;
```

Comment:
1. just to re-confirm, the current `update_range` interface will consider both `range` and `input.vis`, not assuming the stream chunk is compacted(no visibility bitmap)?
2. I think the comments here are not the essential distinction, BatchHashAgg can use the `update` interface with visibility bitmap, and in some cases(such as EOWC) streaming executor can use the `update_range` too.

Reply:
1. Yes

Reply:
1. Yes
2. Right. should be rephrased as "mainly used in"